demo/admin/src/news/handmade/NewsForm.tsx

```diff
@@ -0,0 +1,337 @@
+            {({ values, form }) => (
+                <EditPageLayout>
+                    {saveConflict.dialogs}
+                    <Toolbar>
+                        <ToolbarItem>
+                            <IconButton onClick={stackApi?.goBack}>
+                                <ArrowLeft />
+                            </IconButton>
+                        </ToolbarItem>
+                        <ToolbarTitleItem>
+                            <FormattedMessage id="news.News" defaultMessage="News" />
+                        </ToolbarTitleItem>
+                        <ToolbarFillSpace />
+                        <ToolbarActions>
+                            <FinalFormSaveSplitButton hasConflict={saveConflict.hasConflict} />
+                        </ToolbarActions>
+                    </Toolbar>
+                    <MainContent>
+                        <BlockPreviewWithTabs
+                            previewUrl={`${siteConfig.blockPreviewBaseUrl}/news`}
+                            previewState={{
+                                ...values,
+                                image: DamImageBlock.createPreviewState(values.image, previewContext),
+                                content: NewsContentBlock.createPreviewState(values.content, {
+                                    ...previewContext,
+                                    parentUrl: `${match.url}/content`,
+                                }),
+                            }}
+                            previewApi={previewApi}
+                        >
+                            {[
+                                {
+                                    key: "basic",
+                                    label: (
+                                        <AdminTabLabel
+                                            isValid={() => {
+                                                // TODO remove once tabs aren't unmounted anymore
+                                                if (!form.getFieldState("slug")) {
+                                                    return true;
+                                                }
+
+                                                // TODO is this a good idea?
+                                                return (
+                                                    (form.getFieldState("slug")?.valid &&
+                                                        form.getFieldState("title")?.valid &&
+                                                        form.getFieldState("status")?.valid &&
+                                                        form.getFieldState("date")?.valid &&
+                                                        form.getFieldState("category")?.valid &&
+                                                        form.getFieldState("image")?.valid) ??
+                                                    false
+                                                );
```

Comment:
Why do we want to validate a subset of fields? Shouldn't all fields always be valid?

And: Can we simply use `form.getState().valid` instead of checking for every field?

Reply:
> Why do we want to validate a subset of fields? Shouldn't all fields always be valid?

To show the error correctly for a single tab